app/form_objects/initial_request_form.rb

```diff
@@ -20,4 +21,16 @@ def add_no_results_error
   def provider_search?
     training_provider_code == "-1"
   end
+
+  def selected_number_of_places
+    return if number_of_places.nil?
+
+    if number_of_places_invalid?
+      errors.add(:number_of_places, "You must enter a number")
+    end
+  end
+
+  def number_of_places_invalid?
+    number_of_places == "" || !/\A\d+\z/.match(number_of_places) || number_of_places.to_i < 1
```

Comment:
would it be better to validate what it should be and not what it should not be? abstractly something has a small set to what it should be and infinite set of what it should not be. i think capturing with whitelist would comprehend better

Reply:
Yep good point, I'll change